docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,197 @@
+Log Centralisation
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workloads running on
+top of it.
+
+It currently provides monitoring service, powered by Prometheus and
+AlertManager, in order to expose metrics and alerts for both control-plane and
+workload-plane components.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information when it comes to understanding the root cause of a failure
+or a performance degradation. Because of the distributed nature of MetalK8s and
+workloads running of top of it, the administrators need some tooling to ease
+the analysis of near real time and past logs, from a central endpoint. As such
+these logs should be stored on the platform, for a configurable period.
+Browsing the logs is accessible through an API and a UI. The UI should ease
+correlation between logs and health/performance KPIs as well as alerts.
+
+For organisations having their own Log centralisation system (like Splunk or
+Elasticsearch), MetalK8s should provide a way to push logs into it.
+
+The following requirements focus on application logs. Audit logs are not part
+of the requirements.
+
+Requirements
+------------
+
+*Lightweight*
+
+A lightweight tool to store and expose logs is required in order to minimize
+the HW footprints (CPU, RAM, Disks):
+
+- limited history: Storing the logs for very large period (3 years) is not
+  something metalK8s needs to provide as a feature. This can be achieved using
+  external log centralisation system.
+
+- stable ingestion: It is important to guaranty stable ingestion of the logs
+  and less important to guaranty stable performances when browsing/searching
+  the logs. However, peak loads related to complex logs queries should not
+  impact the application workloads and deploying log storage and search service
+  on infra nodes might help achieving this isolation.
+
+- stream indexing: It is not required to have automatic indexing of logs
+  content. Instead, the log centralisation service should offer basic features
+  to group/filter logs per tag/metadata defining the log stream.
+
+*Accessible from a central UI/API*
+
+Platform Admin or Storage Admin can visualize logs from all containers from all
+namespaces as well as journal logs, including (kubelet, containerd, salt-minion
+, kernel, initrd, services, etc ...) in Grafana. One can correlate logs and
+metrics or alerts in one single Grafana Dashboard.
```

Comment:
Correlation can be only visual or based on a specific time frame, it does not necessarily mean that the dashboard automatically guess the relation between one specific alert, log line and perf metric but being able to see specific KPIs, alerts and log streams at the same time is already with a big value. Also, it could be interesting to visualize, in the same Volume/PV dashboard, IOPS/latency KPIs and Alerts next to some RAID card logs so that a RAID card log error could be correlated with a drop of disk IO performances (or peak of latency) and this can help to understand the root cause very quickly. I don't want to change the 3rd party dashboard we have today but with the time we could build our own dashboards based on real-life experience and leveraging those 3 types of information (kpi, alerts, logs and later on tracing ...)

Reply:
one single dashboard is ambitious. Do you mean that correlate logs is more in a way of counting error/warn logs in this single dashboard ?

Reply:
> we could build our own dashboards based on real-life experience and leveraging those 3 types of information (kpi, alerts, logs and later on tracing ...)

OK that's what I thought. Maybe reformulate to make this obvious we are not building "one singe dashboard", but some dashboards with one clear purpose (e.g. disk issues investigations), combining views for logs ([Logs panel](https://grafana.com/docs/grafana/latest/panels/visualizations/logs-panel/)), metrics and alerts (using all the usual panels). Traces are really for deep-dive however, so I think keeping them as they are exposed in Grafana 7 (split-screen between the dashboard and the tracing view) is sufficient, no need to get them always visible in a dashboard.